core/src/main/java/org/apache/carbondata/core/metadata/datatype/DecimalConverterFactory.java

```diff
@@ -88,11 +88,34 @@ void fillVector(Object valuesToBeConverted, int size, ColumnVectorInfo info, Bit
 
     DecimalConverterType getDecimalConverterType();
 
+    default int getPrecisionForDimension(CarbonColumnVector vector) {
```

Comment:
I can see, these methods are called from a single place only. Then, why it is required to create these methods ?. Instead, modify caller method like below:
"""
public void initializeValues(int size, CarbonColumnVector vector, ColumnVectorInfo vectorInfo) {
      if (vectorInfo.measure == null) {
        DecimalType decimalType = (DecimalType) vector.getType();
        // complex primitive decimal flow comes as dimension
        this.precision = decimalType.getPrecision();
        this.newMeasureScale = decimalType.getScale();
        this.updatedSize = ColumnVectorInfo.getUpdatedPageSizeForChildVector(vectorInfo, size);
      } else {
        CarbonMeasure carbonMeasure = vectorInfo.measure.getMeasure();
        this.precision = carbonMeasure.getPrecision();
        this.newMeasureScale = carbonMeasure.getScale();
      }
    } """

Reply:
okay